src/main/java/bio/terra/service/snapshot/flight/create/AddCreatedInfoToSnapshotRequestStep.java

```diff
@@ -0,0 +1,48 @@
+package bio.terra.service.snapshot.flight.create;
+
+import bio.terra.service.snapshot.flight.SnapshotWorkingMapKeys;
+import bio.terra.service.snapshotbuilder.SnapshotRequestDao;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import java.util.Objects;
+import java.util.UUID;
+
+public class AddCreatedInfoToSnapshotRequestStep implements Step {
+  private final SnapshotRequestDao snapshotRequestDao;
+  private final UUID snapshotRequestId;
+  private final UUID createdSnapshotId;
+  private final String samGroupCreatedBy;
+
+  public AddCreatedInfoToSnapshotRequestStep(
+      SnapshotRequestDao snapshotRequestDao,
+      UUID snapshotRequestId,
+      UUID snapshotId,
+      String samGroupCreatedBy) {
+    this.snapshotRequestDao = snapshotRequestDao;
+    this.snapshotRequestId = snapshotRequestId;
+    this.createdSnapshotId = snapshotId;
+    this.samGroupCreatedBy = samGroupCreatedBy;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    var workingMap = context.getWorkingMap();
+    String samGroupName =
+        Objects.requireNonNull(
```

Comment:
It's OK to use `requireNonNull()` although we don't typically validate this way inside flights, as this is data generated by an upstream Step, so any missing data is be due to a programmer error and shouldn't occur otherwise.

Reply:
Yeah, I think we've recently been avoiding the null checks when pulling info from the flight map.